sdk/core/azure-core/src/main/java/com/azure/core/http/HttpRequest.java

```diff
@@ -197,7 +224,31 @@ public HttpRequest setBody(byte[] content) {
      * @return this HttpRequest
      */
     public HttpRequest setBody(Flux<ByteBuffer> content) {
+        if (content != null) {
+            setBody(BinaryDataHelper.createBinaryData(new FluxByteBufferContent(content)));
```

Comment:
Opposite comment from the above, this doesn't seem to need to call into `setBody(BinaryData)` as it won't set the Content-Length (unless `FluxByteBufferContent` is eagerly reading the `Flux<ByteBuffer>`).